turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -444,6 +444,33 @@ pub(crate) async fn analyse_ecmascript_module_internal(
         referenced_package_json,
     } = *module.determine_module_type().await?;
 
+    let compile_time_info = {
+        let mut free_var_references = compile_time_info.free_var_references.await?.clone_value();
+        let (exports_type, module_typeof, require_typeof) = match specified_type {
+            SpecifiedModuleType::Automatic | SpecifiedModuleType::CommonJs => {
+                ("object", "object", "function")
+            }
+            SpecifiedModuleType::EcmaScript => ("undefined", "undefined", "undefined"),
```

Comment:
That might not be safe to define as undefined. There could still be an unrelated global that exist.

Also I think `require` is also available in ESM in Turbopack

Reply:
I've removed that undefined branch.